client/scripts/Player.js

```diff
@@ -0,0 +1,284 @@
+			node.innerHTML = `${index + 1}. ${player}`
+			if (player == this.socket.id) {
+				node.style.fontWeight = "bold"
+			}
+			target.append(node)
+		})
+	}
+	assignNumber(assigned_number) {
+		this.number = assigned_number
+	}
+	updateImageFromServer(whiteboard_json_data) {
+		this.pad.loadJSON(
+			JSON.parse(decodeURIComponent(whiteboard_json_data.new_board_state))
+		)
+	}
+	displayRoomID(room_id) {
+		document.querySelector(
+			"#room_id"
+		).innerHTML = `Your Room Code is ${room_id}`
+	}
+	// converts entire sketchpad to JSON
+	// then exports to server on submit
+	submitPadToServer() {
+		const newBoardState = JSON.stringify(this.pad.toJSON())
+		this.socket.emit("gameplay_stroke", {
+			player_number: this.number,
+			new_board_state: encodeURIComponent(newBoardState),
+		})
+		// disable player actions until
+		// it is current client's turn again
+		this.disableActions()
+	}
+	getPlayerColorFromNumber() {
+		const colors = [
+			"641E16", // dark red
+			"512E5F", // dark purple
+			"154360", // dark blue
+			"AED6F1", // light blue
+			"145A32", // dark green
+			"82E0AA", // light green
+			"7D6608", // dark yellow
+			"E67E22", // orange
+			"99A3A4", // grey
+			"F5B7B1", // light red
+		]
+
+		return "#" + colors[this.number]
+	}
+	sketchpadInit() {
+		requirejs(["responsive-sketchpad/sketchpad"], Sketchpad => {
+			const sketchpadDiv = this.board.querySelector("#sketchpad")
+			this.pad = new Sketchpad(sketchpadDiv, {
+				line: {
+					size: 5,
+				},
+				onDrawEnd: () => {
+					this.pad.setReadOnly(true)
+					this.board.querySelector("#undo").disabled = false
+					this.board.querySelector("#submit_stroke").disabled = false
+				},
+			})
+
+			const enableSubmit = () => {
+				this.board.querySelector("#submit_stroke").disabled = false
+			}
+
+			const disableSubmit = () => {
+				this.board.querySelector("#submit_stroke").disabled = true
+			}
+
+			const enableRedo = () => {
+				this.board.querySelector("#redo").disabled = false
+			}
+
+			const disableRedo = () => {
+				this.board.querySelector("#redo").disabled = true
+			}
+
+			const enableUndo = () => {
+				this.board.querySelector("#undo").disabled = false
+			}
+
+			const disableUndo = () => {
+				this.board.querySelector("#undo").disabled = true
+			}
+
+			// undo
+			const undo = () => {
+				this.pad.undo()
+				this.pad.setReadOnly(false)
+			}
+			this.board.querySelector("#undo").onclick = function () {
+				undo()
+				disableUndo()
+				enableRedo()
+				disableSubmit()
+			}
+
+			// redo
+			const redo = () => {
+				this.pad.redo()
+				this.pad.setReadOnly(true)
+			}
+			this.board.querySelector("#redo").onclick = function () {
+				redo()
+				disableRedo()
+				enableUndo()
+				enableSubmit()
+			}
+
+			// resize
+			window.onresize = e => {
+				this.pad.resize(sketchpadDiv.offsetWidth)
+			}
+			this.pad.setReadOnly(true)
+		})
+	}
+	createBoard() {
+		const board = document.createElement("div")
```

Comment:
cleannn

in terms of function naming, its hard to see the diff between
createBoard and sketchpadInit